www/api/class/centreon_topcounter.class.php

```diff
@@ -204,19 +208,53 @@ public function getUser()
             $this->soundNotificationsEnabled = true;
         }
 
-        /* Get autologinkey */
-        $query = 'SELECT contact_autologin_key FROM contact WHERE contact_id = ' . (int)$user->user_id;
+        /* Is the autologin feature enabled ? */
+        $query = 'SELECT value FROM options WHERE options.key = "enable_autologin"';
 
         try {
             $res = $this->pearDB->query($query);
         } catch (\Exception $e) {
             throw new \RestInternalServerErrorException('Error getting the user.');
         }
 
-        if ($res->rowCount() === 0) {
-            throw new \RestUnauthorizedException('User does not exists.');
+        if ($row = $res->fetch()) {
+            $enableAutoLogin = $row['value'] == '1' ? true : false;
```

Comment:
As $enableAutoLogin and $enableAutoLoginShortcut are used only once, you could remove them.
And rename the 2 $row (lines 220 and 233) with something like $rowEnableShortcut and $rowEnableAutoLogin instead